hudi-hadoop-mr/src/main/java/org/apache/hudi/hadoop/HoodieParquetInputFormat.java

```diff
@@ -250,6 +122,61 @@ protected FileSplit makeSplit(Path file, long start, long length,
     return split;
   }
 
+  private RecordReader<NullWritable, ArrayWritable> getRecordReaderInternal(InputSplit split,
+                                                                            JobConf job,
+                                                                            Reporter reporter) throws IOException {
+    return mapredParquetInputFormat.getRecordReader(split, job, reporter);
```

Comment:
I see that `MapredParquetInputFormat::getRecordReader` references member variable `realInput` inside and it is passed in through one constructor variant, whereas this class uses the constructor with empty arg.  Do we need to consider the following as well?
```
protected MapredParquetInputFormat(final ParquetInputFormat<ArrayWritable> inputFormat) {
    this.realInput = inputFormat;
    vectorizedSelf = new VectorizedParquetInputFormat();
  }
```

Reply:
Empty ctor actually will just channel into the other one: 

```
public MapredParquetInputFormat() {
    this(new ParquetInputFormat<ArrayWritable>(DataWritableReadSupport.class));
  }

  protected MapredParquetInputFormat(final ParquetInputFormat<ArrayWritable> inputFormat) {
    this.realInput = inputFormat;
    vectorizedSelf = new VectorizedParquetInputFormat();
  }
```

Empty ctor is what this class was using before as well when it was inheriting from `MapredParquetInputFormat` directly

Reply:
Sg.